src/target/riscv/riscv-013.c

```diff
@@ -3424,114 +3412,189 @@ static void log_mem_access_result(struct target *target, bool success, int metho
 		LOG_TARGET_DEBUG(target, "%s", msg);
 }
 
-static bool mem_should_skip_progbuf(struct target *target, target_addr_t address,
-		uint32_t size, bool is_read, char **skip_reason)
+typedef enum {
+	MEM_ACCESS_STATUS_KIND_OK,
+	MEM_ACCESS_STATUS_KIND_DISABLED,
+	MEM_ACCESS_STATUS_KIND_SKIPPED,
+	MEM_ACCESS_STATUS_KIND_FAILED
+} mem_access_status_kind_t;
```

Comment:
If you like: `result` is more descriptive than `status`.

```suggestion
typedef enum {
	MEM_ACCESS_RESULT_KIND_OK,
	MEM_ACCESS_RESULT_KIND_DISABLED,
	MEM_ACCESS_RESULT_KIND_SKIPPED,
	MEM_ACCESS_RESULT_KIND_FAILED
} mem_access_result_kind_t;
```

Reply:
Also you may consider `_RESULT_TYPE_` instead of `_RESULT_KIND_` (but I don't feel strongly).

Reply:
Addressed